[quant][graphmode][fx][bc-breaking] Support for reference pattern for fixqparam ops in eval mode

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #62861
* #62698
* __->__ #62608

Summary:
Insert extra fixeqparam fake quant in the output of fixed qparam ops in fbgemm e.g. sigmoid
so that we can produce reference patterns for these ops

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30053978](https://our.internmc.facebook.com/intern/diff/D30053978)

jerryzh168 committed:
[quant][graphmode][fx][bc-breaking] Support for reference pattern for fixqparam ops in eval mode

Summary:
Insert extra fixeqparam fake quant in the output of fixed qparam ops in fbgemm e.g. sigmoid
so that we can produce reference patterns for these ops

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

jerryzh168 committed:
Update on "[quant][graphmode][fx][bc-breaking] Support for reference pattern for fixqparam ops in eval mode"


Summary:
Insert extra fixeqparam fake quant in the output of fixed qparam ops in fbgemm e.g. sigmoid
so that we can produce reference patterns for these ops

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30053978](https://our.internmc.facebook.com/intern/diff/D30053978)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

jerryzh168 committed:
Update on "[quant][graphmode][fx][bc-breaking] Support for reference pattern for fixqparam ops in eval mode"


Summary:
Insert extra fixeqparam fake quant in the output of fixed qparam ops in fbgemm e.g. sigmoid
so that we can produce reference patterns for these ops

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30053978](https://our.internmc.facebook.com/intern/diff/D30053978)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

jerryzh168 committed:
Update on "[quant][graphmode][fx][bc-breaking] Support for reference pattern for fixqparam ops in eval mode"


Summary:
Insert extra fixeqparam fake quant in the output of fixed qparam ops in fbgemm e.g. sigmoid
so that we can produce reference patterns for these ops

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30053978](https://our.internmc.facebook.com/intern/diff/D30053978)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

jerryzh168 committed:
Update on "[quant][graphmode][fx][bc-breaking] Support for reference pattern for fixqparam ops in eval mode"


Summary:
Insert extra fixeqparam fake quant in the output of fixed qparam ops in fbgemm e.g. sigmoid
so that we can produce reference patterns for these ops

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30053978](https://our.internmc.facebook.com/intern/diff/D30053978)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

jerryzh168 committed:
Update on "[quant][graphmode][fx][bc-breaking] Support for reference pattern for fixqparam ops in eval mode"


Summary:
Insert extra fixeqparam fake quant in the output of fixed qparam ops in fbgemm e.g. sigmoid
so that we can produce reference patterns for these ops

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30053978](https://our.internmc.facebook.com/intern/diff/D30053978)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

raghuramank100 commented on test/quantization/fx/test_numeric_suite_fx.py:
```diff
@@ -460,6 +460,7 @@ def forward(self, x0):
         self.assert_types_for_matched_subgraph_pairs(results, expected_types, mp, mq)
 
     @skipIfNoFBGEMM
+    @unittest.skip("Broken by https://github.com/pytorch/pytorch/pull/62608, need dtype inference support")
```
What is the ETA for dtype inference support?

jerryzh168 replied:
@vkuzo will be working on that, I think maybe after he has some time, not sure about the ETAs..

jerryzh168 committed:
Update on "[quant][graphmode][fx][bc-breaking] Support for reference pattern for fixqparam ops in eval mode"


Summary:
Insert extra fixeqparam fake quant in the output of fixed qparam ops in fbgemm e.g. sigmoid
so that we can produce reference patterns for these ops

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30053978](https://our.internmc.facebook.com/intern/diff/D30053978)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

jerryzh168 committed:
Update on "[quant][graphmode][fx][bc-breaking] Support for reference pattern for fixqparam ops in eval mode"


Summary:
Insert extra fixeqparam fake quant in the output of fixed qparam ops in fbgemm e.g. sigmoid
so that we can produce reference patterns for these ops

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30053978](https://our.internmc.facebook.com/intern/diff/D30053978)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

jerryzh168 committed:
Update on "[quant][graphmode][fx][bc-breaking] Support for reference pattern for fixqparam ops in eval mode"


Summary:
Insert extra fixeqparam fake quant in the output of fixed qparam ops in fbgemm e.g. sigmoid
so that we can produce reference patterns for these ops

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30053978](https://our.internmc.facebook.com/intern/diff/D30053978)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).
